### ISSUE TITLE
`hook.app` is undefined in Event filters

### ISSUE THREAD

**Issue by sscaff1** [feathersjs/feathers#405]
I'm using sequelize. Here is an example scenario of when I need to use this functionality. I have users in groups. When users send a message, I need to see if they are part of that group before allowing them to see the data. GroupUsers is a mapping table mapping groups to users.

My event filter is below. If you `console.log(hook.app)`, it logs `undefined`.
```
export default function () {
  return function filterGroupId(data, connection, hook) {
    const query = {
      userId: connection.user.id,
      groupId: data.groupId,
      deleted: false,
    };
    return hook.app.service('groupUsers').find({ query })
    .then(result => {
      if (result.total > 0) {
        return data;
      }
      return false;
    });
  };
}
```

**Comment by kaiquewdev** [comment]
Your service was using this hook, i mean calling it?

**Comment by sscaff1** [comment]
Yea I have a few before and after hooks running fine before this filter happens.

**Comment by kaiquewdev** [comment]
Solving that issue with an member of the core development team.

https://github.com/feathersjs/feathers/pull/406

**Comment by sscaff1** [comment]
Awesome thank you.

**Comment by daffl** [comment]
Closed via #406
